### PR TITLE
[refactor-frontend] refactor frontend (project structure, redux code, integration with oicd)

### PR DESCRIPTION
**Pull request by tudorgergely**
### What does it fix?

Looked at the frontend code and it looked like it needed some love.

Fixed:
- Integration with OICD is not nicer (let it handle user save inside sessionStorage and other things).
- Use redux with hooks (integrates nicer, can be used in hooks as well)
- Use react hooks where possible.
- Refactor components here are there (removed some useless ones)
- Extract api to separate files (which can be extended when we will add more api calls)

### How has it been tested?

Open, login, logout, enter each route etc

### CODE REVIEW

File: frontend/src/api/api.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+import { getUserToken } from "./auth";
+
+const api = axios.create({
+  baseURL: `${process.env.REACT_APP_IDP_URL}/api`
```
Comment (adascaliteiradu) [inline frontend/src/api/api.js]: It is not REACT_APP_IDP_URL. It should be another REACT_APP_ environment variable for API url

Comment (tudorgergely) [reply frontend/src/api/api.js]: what is the API URL then? because locally `REACT_APP_IDP_URL='http://localhost:5001'` is the correct URL.

Comment (adascaliteiradu) [reply frontend/src/api/api.js]: The API URL is http://localhost:5008
http://localhost:5001 is the IdentityServer URL

Comment (tudorgergely) [reply frontend/src/api/api.js]: sounds good, thanks for clarifying !

**Review by surdu — DISMISSED** [review]

**Review by surdu — APPROVED** [review]

**Pull request merged**